backend/api/oauth.py

```
@@ -0,0 +1,142 @@
+    cursor.execute("SELECT UserID, Role, SourceID FROM Users WHERE Email = ?", (email,))
+    row = cursor.fetchone()
+
+    if row:
+        user_id, role, source_id = row
+        token = create_jwt_token(user_id, email)
+        return RedirectResponse(f"http://localhost:8000/dashboard?token={token}", status_code=303)
+
+    # If user doesn't exist, redirect to signup completion
+    return RedirectResponse(f"http://localhost:8000/html/login.html?email={email}&provider={provider}")
+
+@router.post("/auth/oauth/complete-signup")
```

Flake8 (7.2.0)

[error] expected 2 blank lines, found 1

(E302)